lib/jekyll/readers/theme_assets_reader.rb

```diff
@@ -29,7 +29,7 @@ def read_theme_asset(path)
           Jekyll::Page.new(site, base, dir, name)
       else
         append_unless_exists site.static_files,
-          Jekyll::StaticFile.new(site, base, dir, name)
+          Jekyll::StaticFile.new(site, base, "/#{dir}", name)
```

Comment:
> ?

@parkr 
A `Jekyll::StaticFile` path in `source_dir` : `/assets/img/logo.png`

A `Jekyll::StaticFile` path in `theme_dir` **without this PR** : `assets/images/hero.jpg`
A `Jekyll::StaticFile` path in `theme_dir` **with this PR** : `/assets/images/hero.jpg`

The above minor discrepancy is what causes #5676